### ISSUE TITLE
 Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.

### ISSUE THREAD

**Issue by ryancode-commit** [tonyofrancis/Fetch#638]
Fetch there is an error in compileSdkVersion 31 and above & targetSdkVersion 31 and above
because DefaultFetchNotificationManager there is a Pending intent with Flag_current_update

**Comment by 557711** [comment]
Kindly Solve this issuse Owner of library

**Comment by shreyaag1** [comment]
Please update the library!

**Comment by OmidNejadabbasi** [comment]
@shreyaag1 @ryancode-commit @557711 

I managed to fix the bug by copying and pasting the code into my own class file and corrected the PendingIntent flag.
Here is the link to my code : https://gist.github.com/OmidNejadabbasi/ccda264acc68a65be6004199344d168f

Remember that you should instantiate and pass this object to setNotificationManager() method.

**Comment by sanjaysingh1990** [comment]
Thank you for  this fix

**Comment by ryancode-commit** [comment]
> @shreyaag1 @ryancode-commit @557711 
> 
> I managed to fix the bug by copying and pasting the code into my own class file and corrected the PendingIntent flag.
> Here is the link to my code : https://gist.github.com/OmidNejadabbasi/ccda264acc68a65be6004199344d168f
> 
> Remember that you should instantiate and pass this object to setNotificationManager() method.

I can't paste it
Make a new class or edit DefaultFetchNotigicationManager?

**Comment by OmidNejadabbasi** [comment]
> > @shreyaag1 @ryancode-commit @557711
> > I managed to fix the bug by copying and pasting the code into my own class file and corrected the PendingIntent flag.
> > Here is the link to my code : https://gist.github.com/OmidNejadabbasi/ccda264acc68a65be6004199344d168f
> > Remember that you should instantiate and pass this object to setNotificationManager() method.
> 
> I can't paste it Make a new class or edit DefaultFetchNotigicationManager?

Yes, you should create your own .java file and paste my code and then use that class. I have tested it and it works!

**Comment by sanjaysingh1990** [comment]
This fix worked for me as well after some fixes like string and drawable resources linking.

**Comment by ryancode-commit** [comment]
> > > @shreyaag1 @ryancode-commit @557711
> > > I managed to fix the bug by copying and pasting the code into my own class file and corrected the PendingIntent flag.
> > > Here is the link to my code : https://gist.github.com/OmidNejadabbasi/ccda264acc68a65be6004199344d168f
> > > Remember that you should instantiate and pass this object to setNotificationManager() method.
> > 
> > I can't paste it Make a new class or edit DefaultFetchNotigicationManager?
> 
> > Yes, you should create your own .java file and paste my code and then use that class. I have tested it and it works!

For import net.omidn.fetchme.R.drawable and import net.omidn.fetchme.R.string 
Do you use library? Or what?

**Comment by sanjaysingh1990** [comment]
replace these drawable and string resources with your local resources.

**Comment by ryancode-commit** [comment]
Thank you so much.
